gaps/gap-5_payload_manifest/gap-5_payload_manifest.md

```diff
@@ -0,0 +1,127 @@
+---
+gap: 5
+title: Computation Payload Manifest
+description: Replacement for the `golem.srv.comp.task_package` property
+author: mf (@mfranciszkiewicz)
+status: Draft
+type: Feature
+requires (*optional): <GAP number TBD>
+---
+
+## Abstract
+
+The GAP proposes a new `golem.srv.comp.payload` property aiming to replace the `golem.srv.comp.task_package`. 
+The `payload` namespace covers URLs of computation payloads for multiple hardware architectures, an optional Computation
+Manifest ([GAP](https://github.com/golemfactory/golem-architecture/pull/27)) and also an optional Payload Manifest 
+signature.
+
+## Motivation
+
+The main motivation behind the proposed changes is to provide a source of trust for the executed Payload. The Manifest 
+may include a default, safe set of actions Requestors can execute on a Provider's machine, approved by a trusted 
+Authority. This source of trust allows for implementing high-risk features, such as Internet access for Runtimes, where 
+all traffic is routed via Provider's OS. In order to prevent unwanted actions in a public network on Provider's behalf, 
+she must remain in control of all the actions executed by the Requestor.
+
+The secondary objective is to provide multiple locations of Payloads, targeting different hardware architectures and 
+meeting specific software constraints.
+
+None of the above are possible with the `golem.srv.comp.task_package` property.
+
+## Specification
+
+Computation Payload Manifest signatures are verified by either the Provider Agent, the ExeUnit Supervisor or both.
+Payload and Computation manifests are not expected to have constraints put on them. 
+
+### `golem.srv.comp.payload: String`
```

Comment:
I'm not sure about putting whole manifest in Agreement. I would probably prefer link to manifest and signatures to verify it. If we can't put constraints on these properties, than having this manifest here, doesn't add any value

Reply:
It reduces the complexity of this solution.

- no download issues and delays caused by retries; manifests could be used in negotiations without slowing down the process
- Provider Agent does not need to support a set of transfer protocols when HTTP is no longer sufficient
- no need for separate manifest storage / cache
- the contents of the manifest are always known in case of conflict resolution (expiring links and no manifest in storage)
- less prerequisites for debugging / development
- less interface / CLI changes (e.g. in ExeUnit)
- pivoting from an encoded string to a set of (constrainable) properties will not make this extra code obsolete

Reply:
Marking as 'resolved' after decision on including manifests in Agreements.